Doctrine/Phpcr/SimpleUrlInformationGuesser.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2015 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\SeoBundle\Doctrine\Phpcr;
+
+use Symfony\Cmf\Bundle\SeoBundle\AlternateLocaleProviderInterface;
+use Symfony\Cmf\Bundle\SeoBundle\Model\UrlInformation;
+use Symfony\Cmf\Bundle\SeoBundle\SeoPresentation;
+use Symfony\Cmf\Bundle\SeoBundle\Sitemap\UrlInformationGuesserInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * A base guesser to extract url information from a document persisted with the phpcr-odm.
+ *
+ * @author Maximilian Berghoff <Maximilian.Berghoff@gmx.de>
+ */
+class SimpleUrlInformationGuesser implements UrlInformationGuesserInterface
+{
+    /**
+     * @var AlternateLocaleProviderInterface
+     */
+    protected $alternateLocaleProvider;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var string
+     */
+    private $defaultChangeFrequency;
+
+    /**
+     * @var SeoPresentation
+     */
+    private $seoPresentation;
+
+    /**
+     * @param RouterInterface $router
+     * @param SeoPresentation $seoPresentation
+     * @param $defaultChangeFrequency
+     */
+    public function __construct(
+        RouterInterface $router,
+        SeoPresentation $seoPresentation,
+        $defaultChangeFrequency
+    ) {
+        $this->router = $router;
+        $this->seoPresentation = $seoPresentation;
+        $this->defaultChangeFrequency = $defaultChangeFrequency;
+    }
+
+    /**
+     * @param AlternateLocaleProviderInterface $alternateLocaleProvider
+     */
+    public function setAlternateLocaleProvider(AlternateLocaleProviderInterface $alternateLocaleProvider)
+    {
+        $this->alternateLocaleProvider = $alternateLocaleProvider;
+    }
+
+    /**
+     * {@inheritDocs}
+     */
+    public function guessValues(UrlInformation $urlInformation, $object, $sitemap = 'default')
+    {
+        $urlInformation->setLocation($this->router->generate($object, array(), true));
```

Comment:
yeah, this would be the all-out complex solution. it is more powerful and puts less logic into single guessers, as they would not need to check if a field is set already or not. but it adds yet more things. but we could go that way. not sure if there is any reusable guesser infrastructure we could take or whether we would need to completely implement things here.